src/services/ldap/strategies/general.js

```diff
@@ -0,0 +1,115 @@
+const request = require('request-promise-native');
+
+const AbstractLDAPStrategy = require('./interface.js');
+
+/**
+ * General LDAP functionality
+ * @implements {AbstractLDAPStrategy}
+ */
+class GeneralLDAPStrategy extends AbstractLDAPStrategy {
+	constructor(app, config) {
+		super(app, config);
+	}
+
+	/**
+     * @public
+     * @see AbstractLDAPStrategy#getSchoolsQuery
+     * @returns {Array} Array of Objects containing ldapOu (ldap Organization Path), displayName
+     * @memberof GeneralLDAPStrategy
+     */
+	getSchools() {
+		return Promise.resolve([{
+			displayName: this.config.providerOptions.schoolName,
+			ldapOu: this.config.rootPath,
+		}]);
+	}
+
+	/**
+     * @public
+     * @see AbstractLDAPStrategy#getUsers
+     * @returns {Array} Array of Objects containing email, firstName, lastName, ldapDn, ldapUUID, ldapUID,
+     * (Array) roles = ['teacher', 'student', 'administrator']
+     * @memberof GeneralLDAPStrategy
+     */
+	getUsers(school) {
+		const options = {
+			filter: 'objectClass=person',
+			scope: 'sub',
+			attributes: [
+				this.config.providerOptions.userAttributeNameMapping.givenName,
+				this.config.providerOptions.userAttributeNameMapping.sn,
+				this.config.providerOptions.userAttributeNameMapping.dn,
+				this.config.providerOptions.userAttributeNameMapping.uuid,
+				this.config.providerOptions.userAttributeNameMapping.uid,
+				this.config.providerOptions.userAttributeNameMapping.mail,
+				this.config.providerOptions.userAttributeNameMapping.role,
```

Comment:
Does `attributes: Object.keys(this.config.providerOptions.userAttributeNameMapping),` make sense here? Are there more attributes in the mapping than needed here?